<?xml version="1.0"?>
<launch>

    <node pkg="duda_go" type="new_pid.py" name="lpid_velocity">
        <remap from="wheel" to="lwheel" />
        <remap from="motor_cmd" to="lmotor_cmd" />
        <remap from="wheel_vtarget" to="lwheel_vtarget" />
        <remap from="wheel_vel" to="lwheel_vel" />
        <rosparam param="Kp">15</rosparam>
        <rosparam param="Ki">110</rosparam>
        <rosparam param="Kd">0.001</rosparam>
        <rosparam param="sample_size">3</rosparam>
        <rosparam param="rate">100</rosparam>
    </node>

    <node pkg="duda_go" type="new_pid.py" name="rpid_velocity">
        <remap from="wheel" to="rwheel" />
        <remap from="motor_cmd" to="rmotor_cmd" />
        <remap from="wheel_vtarget" to="rwheel_vtarget" />
        <remap from="wheel_vel" to="rwheel_vel" />
        <rosparam param="Kp">15</rosparam>
        <rosparam param="Ki">100</rosparam>
        <rosparam param="Kd">0.001</rosparam>
        <rosparam param="sample_size">3</rosparam>
        <rosparam param="rate">100</rosparam>
    </node>

    <node pkg="rosserial_python" type="serial_node.py" name="encoder" output="screen">
        <rosparam param="port">/dev/ttyUSB0</rosparam>
        <rosparam param="baud">2000000</rosparam>
    </node>

    <node pkg="duda_go" type="twist_to_motors.py" name="twist_to_motors" output="screen">
        <rosparam param="base_width">0.625</rosparam>
    </node>

    <node pkg="duda_go" type="motor_run.py" name="motor_run" output="screen">
    </node>

</launch>
